app/controllers/sessions_controller.rb

```diff
@@ -23,10 +25,32 @@ def create
       render 'new'
     end
   end
-  
+
   def destroy
     log_out if logged_in?
     redirect_to root_url
   end
+  # require 'pry'
+
+  def omniauth
+    # binding.pry
+    uid = request.env['omniauth.auth'][:uid]
+    provider = request.env['omniauth.auth'][:provider]
+    user = get_account(uid, provider)
+    log_in user # luu session
+    redirect_to root_path
+  end
+
+  private
 
+  def get_account(uid, provider)
+    User.find_or_create_by(uid: uid, provider: provider) do |u|
```

Comment:
ở đây provider phải tách thành bảng riêng và belong to User
User  unique theo email, nên cần xử lý case mà login từ google vs facebook với cùng 1 email thì bảng user chỉ lưu 1 người